src/lib/index.d.ts

```diff
@@ -1,48 +1,68 @@
-import GleeAdapter from './adapter.js'
-import GleeClusterAdapter from './cluster.js'
-import GleeConnection from './connection.js'
-import Glee from './glee.js'
+import GleeAdapter from "./adapter.js";
+import GleeClusterAdapter from "./cluster.js";
+import GleeConnection from "./connection.js";
+import Glee from "./glee.js";
 
-type WebSocketServerType = 'native' | 'socket.io'
+type WebSocketServerType = "native" | "socket.io";
 
 export type GleeClusterAdapterConfig = {
-  adapter?: string | typeof GleeClusterAdapter,
-  name?: string,
-  url: string
+  adapter?: string | typeof GleeClusterAdapter;
+  name?: string;
+  url: string;
+};
+
+export type WebsocketAdapterConfig = {
+  server?: {
+    httpServer?: any
+    adapter?: WebSocketServerType | typeof GleeAdapter
+    port?: number
+  }
+  client?: {
+    authentication?: {
+      http: string
+    }
+  }
 }
 
-export type GleeConfig = {
-  websocket?: {
-    httpServer?: any,
-    adapter?: WebSocketServerType | typeof GleeAdapter,
-    port?: number,
-  },
-  cluster?: GleeClusterAdapterConfig
+export type MqttAdapterConfig = {
+  authentication?: {
+    cert?: string
+    userPassword?: {username: string, password: string}
+    clientId?: string
```

Comment:
I added three different parameters for authentication because I think MQTT allows all three forms of authentication and whatever is specified by the user will be used.